File: app/src/main/java/me/pwcong/jpstart/mvp/presenter/JPStartFragmentPresenterImpl.java
package me.pwcong.jpstart.mvp.presenter;

import android.util.Log;

import java.util.List;

import me.pwcong.jpstart.mvp.bean.JPItem;
import me.pwcong.jpstart.mvp.model.BaseModel;
import me.pwcong.jpstart.mvp.model.JPStartFragmentModelImpl;
import me.pwcong.jpstart.mvp.view.BaseView;
import rx.Subscriber;

/**
 * Created by Pwcong on 2016/9/27.
 */

public class JPStartFragmentPresenterImpl extends BasePresenter<BaseView.JPStartFragmentView>
        implements BasePresenter.JPStartFragmentPresenter {

    private final String TAG = getClass().getSimpleName();

    private BaseModel.JPStartFragmentModel model;

    public JPStartFragmentPresenterImpl(BaseView.JPStartFragmentView view) {
        super(view);

        model = new JPStartFragmentModelImpl();
    }

    @Override
    public void initJPStartFragment(int type) {

        view.setRecyclerView(type);
        model.getData(type, new Subscriber<List<JPItem>>() {
            @Override
            public void onCompleted() {
                Log.i(TAG, "onCompleted: OK");
            }

            @Override
            public void onError(Throwable e) {
                Log.i(TAG, "onError: OK");
            }

            @Override
            public void onNext(List<JPItem> list) {
                view.setData(list);
                Log.i(TAG, "onNext: OK");
            }
        });

    }
}

File: app/src/main/java/com/github/pwcong/jpstart/mvp/presenter/impl/JPStartFragmentPresenterImpl.kt
package com.github.pwcong.jpstart.mvp.presenter.impl

import android.util.Log
import rx.Subscriber

import com.github.pwcong.jpstart.mvp.bean.JPItem
import com.github.pwcong.jpstart.mvp.model.BaseModel.JPStartFragmentModel
import com.github.pwcong.jpstart.mvp.model.impl.JPStartFragmentModelImpl
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter.JPStartFragmentPresenter
import com.github.pwcong.jpstart.mvp.view.BaseView.JPStartFragmentView

class JPStartFragmentPresenterImpl(view: JPStartFragmentView) :
    BasePresenter<JPStartFragmentView>(view),
    JPStartFragmentPresenter {
    private val TAG: String = javaClass.simpleName

    private val model: JPStartFragmentModel = JPStartFragmentModelImpl()

    override fun initJPStartFragment(category: Int) {
        view.setRecyclerView(category)
        model.getData(category, object : Subscriber<List<JPItem>>() {
            override fun onCompleted() {
                Log.i(TAG, "onCompleted: OK")
            }

            override fun onError(e: Throwable?) {
                Log.i(TAG, "onError: OK")
            }

            override fun onNext(list: List<JPItem>) {
                view.setData(list)
                Log.i(TAG, "onNext: OK")
            }
        })
    }
}
